lib/matplotlib/figure.py

```diff
@@ -2031,7 +2026,9 @@ def make_active(event):
 
     def add_axobserver(self, func):
         """Whenever the axes state change, ``func(self)`` will be called."""
-        self._axobservers.append(func)
+        # Connect a wrapper lambda and not func itself, to avoid it being
+        # weakref-collected.
+        self._axobservers.connect(None, lambda arg: func(arg))
```

Comment:
```suggestion
        self._axobservers.connect('_state_change', lambda arg: func(arg))
```
Or '_on_set_as_current' or similar to make clear what this callback is used for.

Reply:
I mildly prefer None because it's impossible to typo it (xref https://github.com/matplotlib/matplotlib/pull/15644#discussion_r365519118), but can change it if you insist.

Reply:
Since this not something the user (or even likely we) will type, typos are not that important here. With that argument we‘d have a larger issue with the whole event system.

If you really care about string typos, make a class-level Attribute of it.

Personally, I find it much more important to give the event that’s connected and processed a proper name.

Reply:
made it a string.